src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -57,6 +58,15 @@ public static DiagnosticData ToDiagnosticData(this Diagnostic diagnostic, Projec
                 return DiagnosticData.Create(project.GetDocument(diagnostic.Location.SourceTree), diagnostic);
             }
 
+            if (diagnostic.Location.Kind == LocationKind.ExternalFile)
```

Comment:
This may affect TypeScript.    Can we check to make sure we don't get duplicated errors for TS issues?  @vladima can also probably show you how to make a project level error to hit this codepath.